packages/java/src/node/java-contribution.ts

```diff
@@ -103,7 +104,7 @@ export class JavaContribution extends BaseLanguageServerContribution {
         this.activeDataFolders.add(dataFolderSuffix);
         clientConnection.onClose(() => this.activeDataFolders.delete(dataFolderSuffix));
 
-        const workspacePath = path.resolve(os.homedir(), '.theia', 'jdt.ls', '_ws_' + dataFolderSuffix);
+        const workspacePath = path.resolve(await this.envServer.getUserDataFolderPath(), 'jdt.ls', '_ws_' + dataFolderSuffix);
```

Comment:
Maybe we can introduce sync counterparts on the backend, for example by using `EnvVariablesServerImpl` here. In this way we don't need to break APIs and convert code to be async. Concrete products will need to rebind `EnvVariablesServerImpl` then, `EnvVariablesServer` is just async API to i.t

Reply:
Don't get your idea completely. I've thought about synchronous implementation for methods in `EnvVariablesServerImpl`, but it implements rpc endpoint interface which requires to have async methods. Can you elaborate on your idea more?

Reply:
Add `EnvVariablesServerImpl.getSyncUserDataFolderPath`, implement `EnvVariablesServer.getUserDataFolderPath` by delegating to the former

Reply:
Yes we can do that, but in such case we need to use specific implementation instead of interface. In many places we already have async methods, so it doesn't bring any problems with using await.

Reply:
ok, let's try to use async versions everywhere to make it simpler